### ISSUE TITLE
provide explicit licensing details for files under input_data

### ISSUE THREAD

**Issue by ghisvail** [AvtechScientific/ASL#15]
Still on the licensing topic, you guys mention in the `COPYRIGHT.md`.

```
Some examples require input files (like .vti or .stl) that can be found in input_data.

    All .stl files were originally obtained from GrabCAD and are subject to their Terms of Service; some of them were modified.
    brain.vti was generated using BrainWeb.
```

This information tells me _where_ the files were obtained, not witch licensing terms these files are in. Please consider giving a short summary of the licensing for both the GrabCAD and BrainWeb data, since both are not straightforward to find.

On my brief reading of the GrabCAD terms of service, I felt that the licensing was very restrictive with regards to redistribution and commercial usage, which does not look compatible with the terms of the `AGPL-3`?

If licensing information cannot be determined accurately, then I will have to strip these data off prior to packaging.

**Comment by AvtechScientific** [comment]
We are in contact with GrabCAD/BrainWeb wrt publishing the data under AGPLv3.

Just in case it will not work out: can we strip the files out of distribution, but download them from our webserver during building with cmake?

**Comment by ghisvail** [comment]
> We are in contact with GrabCAD/BrainWeb wrt publishing the data under AGPLv3.

They will likely ask you to keep their original licensing. As long as it is compatible with the terms of AGPL-3, you guys are fine. Their licensing terms would need to be properly acknowledged in the `COPYRIGHT.md` file though, which is the topic of this issue.

> can we strip the files out of distribution, but download them from our webserver during building with cmake?

In case the files aren't suitable for distribution within your project due to licensing, they will need to be stripped out from your Github repository indeed. You may decide to provide an (optional) download step as part of your CMake.

As far as packaging is concerned on the other hand, we cannot download these additional resources as part of the build process. Packages are built inside a chroot environment with no ability to download remote resources from the internet. Consequently, please make sure that the ASL library can still be built and some of its test suite can be run, if these files are not available.

**Comment by AvtechScientific** [comment]
> As long as it is compatible with the terms of AGPL-3, you guys are fine.

A legal expert is needed to properly determine whether their terms are compatible with AGPL-3, but as you have pointed out there are restriction on "redistribution and commercial usage" which seem to be incompatible.

> Their licensing terms would need to be properly acknowledged in the COPYRIGHT.md file though, which is the topic of this issue.

We have no problem to write with big letters - the files belong to GrabCAD and attach their complete license, but as far as I understand it is not enough for Debian. Correct?

Wrt BrainWeb the things seem to be less complicated, since its us who generated the file, we just used their tool. I think that we are the owners, the same as an image generated by GIMP doesn't belong to GIMP copyright holders, but to the person who created it.

**Comment by ghisvail** [comment]
> A legal expert is needed to properly determine whether their terms are compatible with AGPL-3, but as you have pointed out there are restriction on "redistribution and commercial usage" which seem to be incompatible.

Probably.

> but as far as I understand it is not enough for Debian. Correct?

Anything that does not follow our [guidelines](https://wiki.debian.org/DFSGLicenses) has to be stripped off indeed. From my first read at the GrabCAD licensing, it does not look DFSG compliant.

> I **think** that we are the owners, the same as an image generated by GIMP doesn't belong to GIMP copyright holders, but to the person who created it.

Better be safe than sorry. It might be worth asking whether some kind of acknowledgement is required (publication, funding body...).

**Comment by AvtechScientific** [comment]
1. Brainweb team confirmed that the generated model belongs to us. Acknowledged by linking to their webpage.
2. GrabCAD issue solved by contacting individual authors of the models, who are the copyright holders. Still linking to GrabCAD to give credit to the platform. See COPYRIGHT.md

Can this be closed?

**Comment by ghisvail** [comment]
Please provide explicit licensing _terms_ for the data, that means:
1. Brainweb: explicitly mention that you are the copyright holders of the data (therefore it is AGPL-3)
2. GrabCAB: I still don't know under which terms these individual authors have agreed to redistribute their work ? under AGPL-3 ? something else ? Please copy the relevant licensing terms in `COPYRIGHT.md` in case it is not AGPL-3.

The bottom line is that I should be able to tell, from the `COPYRIGHT.md`, which part of the entire source tree belongs to whom and under which license it is. The former has been correctly addressed so far, but not the latter.

**Comment by AvtechScientific** [comment]
Improved.

**Comment by ghisvail** [comment]
all good, fixed since f0aac0c6266709f7f7b9acabfc1e6d8d4867ff91
